crates/typst/src/layout/align.rs

```diff
@@ -445,6 +476,150 @@ cast! {
     }
 }
 
+/// An internal representation that combines horizontal or vertical alignments. The
+/// allowed alignment positions are designated by the type parameter `H` and `V`.
+///
+/// This is not user-visible, but an internal type to impose type safety. For example,
+/// `SpecificAlignment<HAlignment, OuterVAlignment>` does not allow vertical alignment
+/// position "center", because `V = OuterVAlignment` doesn't have it.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum SpecificAlignment<H, V>
+where
+    H: Into<HAlignment> + Copy,
+    V: Into<VAlignment> + Copy,
+    HAlignment: Into<H>,
+    VAlignment: Into<V>,
+{
```

Comment:
We don't need any of those traits when constructing the enum (perhaps the Copy ones might be needed though, but I think the #[derive(Copy)] macro already takes care of this, and only implements Copy if both parameters are Copy; if it fails to compile, you can force it with `<H: Copy, V: Copy>` instead).

```suggestion
pub enum SpecificAlignment<H, V> {
```

Reply:
Yup, and `Copy` is not necessary either.